File: frontend/src/services/transfers-service.ts
import { API_CONFIG } from '@/config/constants';
import type { Transfer, TransferFormData } from '@/types';

import { BaseService } from './base-service';

class TransfersService extends BaseService<Transfer, TransferFormData> {
  constructor() {
    super(API_CONFIG.ENDPOINTS.TRANSFERS);
  }
}

export const transfersService = new TransfersService();

File: apps/mobile/lib/services/transfers_service.dart
import '../models/transfer.dart';
import 'base_service.dart';

class TransfersService extends BaseService<Transfer> {
  TransfersService(super.client)
      : super(
          resourcePath: '/api/v1/transfers/',
          fromJson: Transfer.fromJson,
          toJson: (transfer) => transfer.toJson(),
        );
}
